week-1-ts-ecosystem/w01e03-object-oriented-structure/src/models/cart/cart.ts

```diff
@@ -0,0 +1,46 @@
+import {ProductType} from '../product/product.types';
+import {Product} from '../product/product';
+
+export class Cart<Type extends ProductType> {
```

Comment:
Samo wykorzystanie tutaj generyków - to dobra droga. Jednak istnienie `ProductType` razem z `Product` spowodowało że Twój kształt danych @gkubiak92 jest "rozproszony" w 2 miejscach

i tak naprawdę i `class` i `interface` robią to samo - pilnują tego samego modelu danych.
W tym układzie jeśli coś byś chciał zmienić w modelu - musisz to zrobić w 2 miejscach.

Zauważ że klasa to reprezentacja za równo danych jak i typu danych.
Powienieneś przejść na np. układ klas z Konretnymi nazwami typów produktu dziedziczącymi po klasie bazowej.